x-pack/plugins/observability_solution/observability_ai_assistant/server/rule_connector/index.ts

```diff
@@ -0,0 +1,248 @@
+
+const ParamsSchema = schema.object({
+  connector: schema.string(),
+  message: schema.string({ minLength: 1 }),
+});
+
+const RuleSchema = schema.object({
+  id: schema.string(),
+  name: schema.string(),
+  tags: schema.arrayOf(schema.string(), { defaultValue: [] }),
+  ruleUrl: schema.nullable(schema.string()),
+});
+
+const AlertStateSchema = schema.oneOf([
+  schema.literal('new'),
+  schema.literal('ongoing'),
+  schema.literal('recovered'),
+  schema.literal('unknown'),
+]);
+
+const ConnectorParamsSchema = schema.object({
+  connector: schema.string(),
+  message: schema.string({ minLength: 1 }),
+  rule: RuleSchema,
+  alertState: AlertStateSchema,
+});
+
+type AlertState = TypeOf<typeof AlertStateSchema>;
+export type ActionParamsType = TypeOf<typeof ParamsSchema>;
+export type ConnectorParamsType = TypeOf<typeof ConnectorParamsSchema>;
+type RuleType = TypeOf<typeof RuleSchema>;
+
+export type ObsAIAssistantConnectorType = ConnectorType<{}, {}, ConnectorParamsType, unknown>;
+
+export type ObsAIAssistantConnectorTypeExecutorOptions = ConnectorTypeExecutorOptions<
+  {},
+  {},
+  ConnectorParamsType
+>;
+
+export function getObsAIAssistantConnectorType(
+  initResources: (request: KibanaRequest) => Promise<ObservabilityAIAssistantRouteHandlerResources>
+): ObsAIAssistantConnectorType {
+  return {
+    id: OBSERVABILITY_AI_ASSISTANT_CONNECTOR_ID,
+    isSystemActionType: true,
+    getKibanaPrivileges: (params) => ['api:observabilityAIAssistant'],
+    minimumLicenseRequired: 'gold',
+    name: i18n.translate('xpack.observabilityAiAssistant.alertConnector.title', {
+      defaultMessage: 'Observability AI Assistant',
+    }),
+    supportedFeatureIds: [AlertingConnectorFeatureId],
+    validate: {
+      config: {
+        schema: schema.object({}),
+        customValidator: () => {},
+      },
+      params: {
+        schema: ConnectorParamsSchema,
+      },
+      secrets: {
+        schema: schema.object({}),
+      },
+    },
+    renderParameterTemplates,
+    executor(options) {
+      return executor(options, initResources);
+    },
+  };
+}
+
+function renderParameterTemplates(
+  logger: Logger,
+  params: ConnectorParamsType,
+  variables: Record<string, unknown>
+): ConnectorParamsType {
+  return {
+    connector: params.connector,
+    message: params.message,
+    rule: params.rule,
+    alertState: params.alertState,
+  };
+}
+
+async function executor(
+  execOptions: ObsAIAssistantConnectorTypeExecutorOptions,
+  initResources: (request: KibanaRequest) => Promise<ObservabilityAIAssistantRouteHandlerResources>
+): Promise<ConnectorTypeExecutorResult<unknown>> {
+  const state = execOptions.params.alertState;
+  if (state !== 'new') {
+    // system connector action frequency can't be user configured. we use this
+    // flag as dedup mechanism to prevent triggering the same worfklow again.
+    // while recovered state is a valid trigger we only allow new alerts for now
```

Comment:
I wanted to limit unwanted side effects by executing the connector when alert is recovered, but maybe documenting that is enough and we can also execute on recovered state. We could also have two text boxes to set instructions when alert goes from off -> on and on -> off ?

Reply:
I think for now (8.14) we can depend on the LLM to figure out and document it. It would be good to clearly state to the LLM in the instructions that the alert has recovered. Definitely something we should take into account though.